packages/grid_client/src/primitives/nodes.ts

```diff
@@ -373,6 +377,72 @@ class Nodes {
       });
   }
 
+  /**
+   * Searches for a disk in the given DisksPool and returns the index of the first occurrence
+   * where the disk size is greater than or equal to the specified disk size.
+   *
+   * @param {number[]} DisksPool - An array containing the sizes of disks in the pool.
+   * @param {number} disk - The size of the disk to search for.
+   * @returns {number} - The index of the first disk in the DisksPool that has a size greater than or equal to the specified disk size.
+   *                    If no disk is found, it returns -1.
+   */
+  findDiskIndex(DisksPool: number[], disk: number) {
+    for (const index in DisksPool) {
+      if (DisksPool[index] >= disk) return +index;
+    }
+    return -1;
+  }
```

Comment:
you don't have to keep track of the index, the `diskPools` are already sent by reference and you can do the subtraction here